### PR TITLE
Feat: static configs in phone module replaced with i18nify-data

### PR DESCRIPTION
**Pull request by tarun-khanna**
# Description
`DIAL_CODE_MAPPER, PHONE_REGEX_MAPPER, PHONE_FORMAT_MAPPER` replaced with i18nify-data json configs for phone number module

## Changes Made

List the main changes made in this pull request.

| Title                  | --- |
| ---------------------- | --- |
| JIRA link              | https://razorpay.atlassian.net/browse/ATLAS-169  |
| Slack thread (if any)  | NA  |
| Product spec           | NA  |
| Tech spec/One-pager    | NA  |
| Bundle Size Difference | NA  |
| Fixes Issue            | NA  |

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This change is a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Is your change backward compatible ?
- [ ] Tested on major browsers (Chrome, Firefox, Safari, IE) ?
- [x] Tested in a consumer application(s) ?

## Additional Notes

Any additional information that would be helpful for the reviewer.

## Checklist:

- [x] Add Jira ID(s) in PR title and in the description?
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My code is written in TypeScript and follows the coding standards of the project.
- [ ] I have added relevant documentation and updated the README if necessary.
- [ ] My commit messages are clear and follow the project's commit message conventions.
- [ ] Is any external library added?
- [ ] My changes do not introduce any new runtime errors or warnings.
- [ ] All relevant unit tests have been added/updated in my PR for the change.
- [ ] Any Screenshots (mobile & desktop) required for PR? If yes, have you added the respective screenshots ?
- [ ] Any manual dev testing done by you on beta/func environment?
- [ ] Reviewer added (SLA 2 days)

## Reviewer Checklist

- [ ] Sufficient QA/Dev-Testing is done with proof (test cases list)
- [ ] Ensure that the change in bundle size falls within the acceptable range.

---

## PR Title Format

Format: `<type>: <subject>`

Types can be as follows:

- `feat`: (new feature for the user, not a new feature for build script)
- `fix`: (bug fix for the user, not a fix to a build script)
- `docs`: (changes to the documentation)
- `style`: (formatting, missing semi colons, etc; no production code change)
- `refactor`: (refactoring production code, eg. renaming a variable)
- `test`: (adding missing tests, refactoring tests; no production code change)
- `chore`: (updating grunt tasks etc; no production code change)

### CODE REVIEW

File: i18nify-data/phone-number/country-code-to-phone-number/data.json [hunk 1]
```diff
@@ -1,1259 +1,1140 @@
...
+      "dial_code": "+60",
+      "format": "xx xxxxx xx",
+      "regex": "^(\\+?6?0)?(\\d{1,3})[-. ]?(\\d{7,8})$"
+    },
     "AF": {
       "dial_code": "+93",
       "format": "xx xxx xxxx",
-      "regex": "/^(?:\\+93)?\\d{9}$/"
-    },
-    "AX": {
```
[the remark this answers is not in the document]
Comment (Chaitanya-019) [reply i18nify-data/phone-number/country-code-to-phone-number/data.json:8]: +1

File: i18nify-data/phone-number/country-code-to-phone-number/data.json [hunk 2]
```diff
@@ -1,1259 +1,1140 @@
...
       "format": "xxx-xxxx",
-      "regex": "/^(?:\\+1-284)?\\d{7}$/"
+      "regex": "^(?:\\+1-284)?\\d{7}$"
     },
     "VI": {
       "dial_code": "+1",
       "format": "xxx-xxxx",
-      "regex": "/^(?:\\+1-340)?\\d{7}$/"
-    },
-    "WF": {
```
Comment (Chaitanya-019) [inline i18nify-data/phone-number/country-code-to-phone-number/data.json:1223]: Why are some countries data is being removed?

File: i18nify-data/phone-number/dial-code-to-country/data.json [hunk 3]
```diff
@@ -1,663 +1,228 @@
 {
   "dial_code_to_country": {
-    "+93": [
```
[the remark this answers is not in the document]
Comment (tarun-khanna) [reply i18nify-data/phone-number/dial-code-to-country/data.json:3]: No, this is just re-ordered when I removed + prefix from the keys. 
We handle keys without this prefix, so had to make this change

File: i18nify-data/phone-number/country-code-to-phone-number/data.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (tarun-khanna) [reply i18nify-data/phone-number/country-code-to-phone-number/data.json:8]: reverted these changes

**Review by Chaitanya-019 — APPROVED** [review]
LGTM

File: .github/workflows/i18nify-js-validate.yml [hunk 4]
```diff
@@ -30,6 +30,8 @@ jobs:
       - uses: actions/setup-node@v2
         with:
           node-version: 20.3.1
+      - name: List files
```
Comment (RgnDunes) [inline .github/workflows/i18nify-js-validate.yml:33]: Why is this required ?

File: packages/i18nify-js/src/modules/phoneNumber/data/phoneRegexMapper.json [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+  "TN": "^(?:(?:\\+216)|22|9[1-9])\\d{7}$",
+  "TR": "^(?:(?:\\+90)|(0))\\s?5\\d{9}$",
+  "TM": "^(?:\\+993)?\\d{8}$",
+  "TC": "^(?:\\+1-649)?\\d{7}$",
+  "TV": "^(?:\\+688)?\\d{5}$",
+  "UG": "^(?:(?:\\+256)|0)?[39]\\d{8}$",
+  "UA": "^(?:(?:\\+380)|(0))?(39|50|63|66|67|68|91|92|93|94|95|96|97|98|99)\\d{7}$",
+  "AE": "^(?:\\+?971|0)?(?:2|3|4|6|7|9)\\d{8}$",
+  "GB": "^(?:(?:\\+44\\s?|0)7\\d{3}(\\s?\\d{4,})?)$",
+  "UM": "",
```
Comment (RgnDunes) [inline packages/i18nify-js/src/modules/phoneNumber/data/phoneRegexMapper.json:237]: Why is this empty ?

File: packages/i18nify-js/src/modules/phoneNumber/data/phoneRegexMapper.json [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+  "SC": "^(?:(?:\\+248)|\\d{4})\\d{5}$",
+  "SL": "^(?:(?:\\+232)|(?:0))?\\d{9}$",
+  "SG": "^(?:(?:\\+65)|(?:\\(\\+65\\))|(?:65))\\d{4}\\d{4}$",
+  "SX": "^(?:\\+1-721)?\\d{7}$",
+  "SK": "^(?:(?:\\+421))?(0|9[0-8])\\d{8}$",
+  "SI": "^(?:(?:\\+386)|0)?[1-59]\\d{7,8}$",
+  "SB": "^(?:\\+677)?\\d{5}$",
+  "SO": "^(?:\\+252|0)?[567]\\d{7}$",
+  "ZA": "^(?:(?:\\+27)|0)(\\d{9})$",
+  "GS": "",
```
Comment (RgnDunes) [inline packages/i18nify-js/src/modules/phoneNumber/data/phoneRegexMapper.json:209]: Why is this empty ?

File: packages/i18nify-js/src/modules/phoneNumber/data/phoneRegexMapper.json [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+  "OM": "^(?:\\+?968)?(?:95|96|97|98)\\d{6}$",
+  "PK": "^(?:(?:\\+92)|0)?[345]\\d{9}$",
+  "PW": "^(?:\\+680)?\\d{7}$",
+  "PS": "^(?:(?:\\+970))(5[2349])\\d{7}$",
+  "PA": "^(?:(?:\\+507)\\s?)?[46]\\d{6,7}$",
+  "PG": "^(?:\\+?675)?(?:[7-9]\\d{7})$",
+  "PY": "^(?:(?:\\+595|0)9[9876]\\d{7})$",
+  "PE": "^(?:(?:\\+51)|0)?(?:9\\d{8})$",
+  "PH": "^(?:(?:\\+?63)|0)(?:\\d{10})$",
+  "PN": "",
```
Comment (RgnDunes) [inline packages/i18nify-js/src/modules/phoneNumber/data/phoneRegexMapper.json:178]: Why is this empty ?

File: packages/i18nify-js/src/modules/phoneNumber/data/phoneRegexMapper.json [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+  "GD": "^(?:\\+1-473)?(?:\\d{7})$",
+  "GP": "^(?:\\+590)?\\d{9}$",
+  "GU": "^(?:\\+1-671)?\\d{7}$",
+  "GT": "^(?:\\+502)?[2468]\\d{7,8}$",
+  "GG": "^(?:\\+44-1481)?\\d{6}$",
+  "GN": "^(?:\\+224)?\\d{8}$",
+  "GW": "^(?:\\+245)?\\d{7}$",
+  "GY": "^(?:(?:\\+592)?(?:(?:\\s)?[2-9])(?:(?:\\s)?\\d))?(?:(?:\\s)?\\d{4})$",
+  "HT": "^(?:\\+?509)?\\d{8}$",
+  "HM": "",
```
Comment (RgnDunes) [inline packages/i18nify-js/src/modules/phoneNumber/data/phoneRegexMapper.json:99]: Why is this empty ?

File: packages/i18nify-js/src/modules/phoneNumber/data/phoneRegexMapper.json [hunk 9]
```diff
@@ -0,0 +1,253 @@
...
+  "SZ": "^(?:\\+?268)?\\d{7,8}$",
+  "ET": "^(?:\\+?251)?[1-59]\\d{8}$",
+  "FK": "^(?:\\+500)?(?:\\d{5})$",
+  "FO": "^(?:\\+298)?(?:\\d{6})$",
+  "FJ": "^(?:(?:\\+?679)?\\s?\\d{3}\\s?\\d{4})?$",
+  "FI": "^(?:\\+358)?(?:\\d{5,12})$",
+  "FR": "^(?:\\+33)?(?:\\d{9})$",
+  "GF": "^(?:\\+594)?\\d{9}$",
+  "PF": "^(?:\\+689)?\\d{6}$",
+  "TF": "",
```
Comment (RgnDunes) [inline packages/i18nify-js/src/modules/phoneNumber/data/phoneRegexMapper.json:81]: Why is this empty ?

File: packages/i18nify-js/src/modules/phoneNumber/data/phoneRegexMapper.json [hunk 10]
```diff
@@ -0,0 +1,253 @@
...
+  "BE": "^(?:\\+32)?(?:\\d{8,9})$",
+  "BZ": "^(?:(?:\\+501)?\\s?)?[622]\\d{4}$",
+  "BJ": "^(?:\\+229)?(?:\\d{8})$",
+  "BM": "^(?:(?:\\+1)?441)?(?:\\d{2})?\\d{7}$",
+  "BT": "^(?:\\+975)?(?:\\d{8})$",
+  "BO": "^(?:(?:\\+|0{0,2})591)?(?:(?:2|3|7|6)\\d{7})$",
+  "BQ": "^(?:\\+599)?(?:\\d{7})$",
+  "BA": "^(?:\\+387)?(?:\\d{6,8})$",
+  "BW": "^(?:(?:\\+267)?\\s?)?[74]\\d{7}$",
+  "BV": "",
```
Comment (RgnDunes) [inline packages/i18nify-js/src/modules/phoneNumber/data/phoneRegexMapper.json:32]: Why is this empty ?

File: packages/i18nify-js/src/modules/phoneNumber/data/phoneRegexMapper.json [hunk 11]
```diff
@@ -0,0 +1,253 @@
+{
+  "AF": "^(?:\\+93)?\\d{9}$",
+  "AX": "^\\+358\\d{5,8}$",
+  "AL": "^(?:\\+?355)?(?:[4-9]\\d{7}|6\\d{8})$",
+  "DZ": "^(?:\\+?213|0)([567]\\d{8})$",
+  "AS": "^(?:\\+1-684)?(?:\\d{7})$",
+  "AD": "^(?:\\+376)?\\d{6}$",
+  "AO": "^(?:\\+244)?(?:\\d{9})$",
+  "AI": "^(?:\\+1-264)?(?:\\d{7})$",
+  "AQ": "",
```
Comment (RgnDunes) [inline packages/i18nify-js/src/modules/phoneNumber/data/phoneRegexMapper.json:10]: Why is this empty ?

**Review by RgnDunes — CHANGES_REQUESTED** [review]

File: .github/workflows/i18nify-js-validate.yml (hunk 4 above)
Comment (tarun-khanna) [reply .github/workflows/i18nify-js-validate.yml:33]: my bad, was experimenting with gh actions.
Will remove.

File: packages/i18nify-js/src/modules/phoneNumber/data/phoneRegexMapper.json (hunk 11 above)
Comment (tarun-khanna) [reply packages/i18nify-js/src/modules/phoneNumber/data/phoneRegexMapper.json:10]: Created this file from i18nify-data which must have this empty as well.

**Review by RgnDunes — APPROVED** [review]
